richards-benchmark/harness.h

```diff
@@ -0,0 +1,34 @@
+/** A simple harness for the C benchmarks
+ *
+ * It is included by every benchmark file and provides the time measuring,
+ * as well as standardized command-line argument parsing.
+ */
+
+#include <sys/time.h>
+
+unsigned long microseconds() {
+  // Not monotonic
+  struct timeval t;
+  gettimeofday(&t, NULL);
```

Comment:
Should we use `clock_gettime`? Admittedly, `CLOCK_MONOTONIC` on Linux isn't actually monotonic in the commonly understood sense and we need to use `CLOCK_MONOTONIC_RAW` on Linux (but not BSD).

Reply:
I didn't change this code, I retained Stefan Marr's original timing harness from https://github.com/smarr/Classic-Benchmarks/

Reply:
Oh! We should reference that (as it's copyrighted by Stefan, or his employers).